nucypher/network/middleware.py

```diff
@@ -102,6 +102,17 @@ def reencrypt(self, work_order):
             cfrags_and_signatures)
         return cfrags
 
+    def revoke_arrangement(self, ursula, revocation_notice):
+        # TODO: Implement revocation receipts
+        response = requests.delete("https://{}/kFrag/{}".format(ursula.rest_interface,
+                                                              revocation_notice.arrangement_id.hex()),
+                                 bytes(revocation_notice), verify=ursula.certificate_filepath)
+        if not response.status_code == 200:
+            if response.status_code == 404:
+                raise RuntimeError("404 - KFrag doesn't exist to revoke with id {}".format(notice.arrangement_id))
```

Comment:
So - this is essentially an endpoint that will reveal weather or not an Ursula has a particular k_frag?

Reply:
There will also be a 404 if someone doesn't have a right to revoke though. And Alice certainly has a right to know if Ursula has this k_frag